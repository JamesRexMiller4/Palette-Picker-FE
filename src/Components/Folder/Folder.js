import React, { useState, useEffect } from 'react';
import { useStore } from '../App/App';
import { getPalettes } from '../../apiCalls/apiCalls';
import PaletteContainer from '../PaletteContainer/PaletteContainer';
import './Folder.scss';

const Folder = ({ id }) => {
  const { state, dispatch } = useStore();
  const [folderState, setFolderState ] = useState([]);
  
  useEffect(() => {
    getPalettes(id)
      .then(data => {
        let lockedDataArr = data.map(element => {
          return (
            {
              id: element.id, 
              palette_name: element.palette_name,
              palette: [ 
                {color: element.color_one, locked: true},
                {color: element.color_two, locked: true},
                {color: element.color_three, locked: true},
                {color: element.color_four, locked: true},
                {color: element.color_five, locked: true},
              ],
              folder_id: element.folder_id
            }
          )
        })
        dispatch({type: 'SET CURRENT PALETTES', payload: lockedDataArr})
      })
      .catch(error => console.error(`Something went wrong ${error}`))
  }, [])

  // useEffect((prevState) => {
  //   if (state.currentPalettes.length > 0 && state !== prevState) {
  //   getPalettes(id)
  //     .then(data => {
  //       let lockedDataArr = data.map(element => {
  //         return (
  //           {
  //             id: element.id, 
  //             palette_name: element.palette_name,
  //             palette: [ 
  //               {color: element.color_one, locked: true},
  //               {color: element.color_two, locked: true},
  //               {color: element.color_three, locked: true},
  //               {color: element.color_four, locked: true},
  //               {color: element.color_five, locked: true},
  //             ],
  //             folder_id: element.folder_id
  //           }
  //         )
  //       })
  //     dispatch({type: 'SET CURRENT PALETTES', payload: data})
  //     })
  //     .catch(error => console.error(`Something went wrong: ${error}`))
  //   }
  // }, [state])

  let palettesArray = state.currentPalettes.map(palette => {
    let paletteArr = Object.values(palette)
    return (
        <div className='palettes-viewer-div' key={palette.id}>
          <PaletteContainer key={palette.id} id={palette.id} 
          palette={palette.palette} name={palette.palette_name}/>
        </div>)
  })


  return (
    <div className='folder'>
      {palettesArray}
    </div>
  )
}

export default Folder;