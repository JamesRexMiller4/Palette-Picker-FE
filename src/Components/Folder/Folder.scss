
.palettes-viewer-div {
  display: flex;
  width: 80%;
  flex-flow: reverse;
  padding: 20px;
  align-items: center;
  justify-content: center;
}

.palette-name-h2 {
  margin-left: 20px;
}