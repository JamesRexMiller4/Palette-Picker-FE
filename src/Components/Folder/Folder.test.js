import React from 'react';
import Folder from './Folder';
import { shallow } from 'enzyme';

describe('App', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<Folder />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})