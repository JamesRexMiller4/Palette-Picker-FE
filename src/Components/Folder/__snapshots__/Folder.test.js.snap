// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `undefined`;

exports[`App should match the snapshot 2`] = `undefined`;
